crates/ty_python_semantic/src/program.rs

```diff
@@ -89,11 +104,48 @@ impl Program {
 #[derive(Clone, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize))]
 pub struct ProgramSettings {
-    pub python_version: PythonVersion,
+    pub python_version: PythonVersionWithSource,
     pub python_platform: PythonPlatform,
     pub search_paths: SearchPathSettings,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub enum ValueSource {
```

Comment:
Nit: Maybe `PythonVersionSource` to make it less ambiguess (identically named structs can be annoying when you do: Go to symbol because they all show up)

Reply:
I wondered if we might start retaining the source for some other settings (e.g. search paths) so that we could also annotate them in diagnostics. But I'll make the renaming you suggest for now (it can easily be renamed in the future if we start using the enum in other contexts)